types/foundry/client/applications/apps/file-picker.d.mts

```diff
@@ -0,0 +1,329 @@
+import type User from "@client/documents/user.mjs";
+import type {
+    ApplicationClosingOptions,
+    ApplicationConfiguration,
+    ApplicationTab,
+    ApplicationTabsConfiguration,
+    FormFooterButton,
+} from "../_types.d.mts";
+import ApplicationV2 from "../api/application.mjs";
+import HandlebarsApplicationMixin, {
+    type HandlebarsRenderOptions,
+    type HandlebarsTemplatePart,
+} from "../api/handlebars-application.mjs";
+
+export type FilePickerSource = "data" | "public" | "s3";
+
+export interface FilePickerConfiguration extends ApplicationConfiguration {
+    /** A type of file to target. Default: `"any"` */
+    type?: (typeof FilePicker.FILE_TYPES)[number];
+    /** A current file source in "data", "public", or "s3". */
+    activeSource?: FilePickerSource;
+    /** A callback function to trigger once a file has been selected */
+    callback?: Function;
+    /** */
+    current?: string;
+    /** A flag which permits explicitly disallowing upload, `true` by default */
+    allowUpload?: boolean;
+    /** An HTML form field that the result of this selection is applied to */
+    field: HTMLElement;
+    /** An HTML button element which triggers the display of this picker */
+    button: HTMLButtonElement;
+    /**  */
+    favorites?: Record<string, FavoriteFolder>;
+    /** The picker display mode in FilePicker.DISPLAY_MODES */
+    displayMode?: string;
+    /** Display the tile size configuration. */
+    tileSize?: boolean;
+    /** Redirect to the root directory rather than starting in the source directory of one of these files. */
+    redirectToRoot?: string[];
+}
```

Comment:
Make all properties of `ApplicationConfiguration` and extensions thereof required unless they might ultimately still be omitted. The core `@typedef` conflates constructor parameters and the complete object that eventually exists.